### PR TITLE
Use ES5 and support running in workers

### PR DESCRIPTION
**Pull request by shvaikalesh**
Replaces #22 and #24.

### CODE REVIEW

File: require.js [hunk 1]
```diff
@@ -85,12 +83,10 @@
   };
 
   require.register = require.define = function(bundle, fn) {
-    if (typeof bundle === 'object') {
-      for (var key in bundle) {
-        if (has.call(bundle, key)) {
-          require.register(key, bundle[key]);
-        }
-      }
+    if (bundle && typeof bundle === 'object') {
```
Comment (shvaikalesh) [inline require.js]: Cause ya know, `typeof null`

File: require.js [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
 
-  var globals = typeof window === 'undefined' ? global : window;
+  var globals = new Function('return this')();
   if (typeof globals.require === 'function') return;
 
   var modules = {};
   var cache = {};
   var aliases = {};
-  var has = ({}).hasOwnProperty;
+  var has = {}.hasOwnProperty;
```
Comment (shvaikalesh) [inline require.js]: `()` are needed only in the beginning of the statement.

File: require.js [hunk 3]
```diff
@@ -36,8 +35,7 @@
   };
 
   var initModule = function(name, definition) {
-    var hot = null;
```
Comment (shvaikalesh) [inline require.js]: Hot or not?

File: require.js (hunk 2 above)
Comment (paulmillr) [inline require.js]: Are you 100% sure about that? There were a few problems in some browsers because of that.

Comment (paulmillr) [inline require.js]: Older IEs etc

Comment (shvaikalesh) [inline require.js]: If that caused syntax errors, `var has = { a: 1 }` would not work either. I am pretty familiar with IE8 (IE9 is fine on ES) bugs, this is not one of them.

**Pull request merged**